migrations/drivers.go

```
@@ -16,95 +16,238 @@ import (
 
 // TODO: use now() from writefreely pkg
 func (db *datastore) now() string {
-	if db.driverName == driverSQLite {
+	switch db.driverName {
+	case driverSQLite:
 		return "strftime('%Y-%m-%d %H:%M:%S','now')"
+	case driverMySQL:
+		return "NOW()"
+	case driverPostgres:
+		return "NOW()"
 	}
-	return "NOW()"
+
+	return "" // placeholder
 }
 
 func (db *datastore) typeInt() string {
-	if db.driverName == driverSQLite {
+	switch db.driverName {
+	case driverSQLite:
 		return "INTEGER"
+	case driverMySQL:
+		return "INT"
+	case driverPostgres:
+		return "INT"
 	}
-	return "INT"
+
+	return "" // placeholder
```

Not sure how this placeholder makes any sense?

---

those placeholders are for potential wrong values in `db.driverName` and make the compiler and/or language server happy. (they may not accept an non-`nil`-returning function returning nothing)

if the code execution reaches those placeholders it indicates something goes seriously wrong, (with configuration) maybe we should `panic()` instead of returning an empty string.

---

Ah, I see, thanks for the insight! I'm no Go expert, but to me it sounds like a panic is more appropriate than generating invalid SQL.

---

ok i gonna create a panic function for such conditions and invoke them instead of returning the placeholder empty string.